### PR TITLE
Model submission for Absolem sensor configurations 3,4,5 by CTU-CRAS-Norlab

### PR DESCRIPTION
**Pull request by peci1**
Depends and builds on top of #877 and #878. If you want to see just changes from the sensor configurations, look e.g. at https://github.com/ctu-vras/subt/compare/submitted_models/ctu_cras_norlab_absolem/remove_annoying_visuals...ctu-vras:submitted_models/ctu_cras_norlab_absolem_improved .

The Absolem robots are just now undergoing a facelift which removes the tilting 2D lidar, adds a 3D lidar and makes necessary changes to the shape and weight distribution of the body. Additionally, we have identified and found a way to fix a source of very strong vibrations when the robot was driving 0.5 - 0.6 m/s. It is mechanically capable of driving using this velocity, but the vibrations made the sensor data unusable. After fixing the vibrations, it is okay to drive 0.5 or maybe even 0.6 m/s. So I increased the velocity limit for sensor configs 3, 4 and 5 to 0.6 m/s (specified in `sensor_config_3/launch/common.rb`). I didn't touch the weight of the body as it should be more or less the same as before (we remove a 2.5 kg lidar and exchange it for a 0.5 kg one, but we extend the body a bit, so it should cancel out). What will change is weight distribution and now the center of gravity will be much closer to the track hinges.

The facelift has deadline in 7 weeks, so if any validation data with the real robot are required, we will provide them later. All sensors are the same as on our MARV robot, so I suggest assuming they do not need explicit validation in this PR. I managed to fix some of the problems @adlarkin discovered on MARV.

This PR touches spawner.rb and vehicle_topics.launch of sensor config 1 quite heavily, but all changes are in the name of better maintainability and "extensibility" by other sensor configs. I explicitly verified that the changes did not break anything on sensor configs 1 and 2 (all data are coming to ROS, I can drive, I can control flippers, I can drop breadcrumbs). The changes to IMU and wheel slip made by SubT organizers were retained and verified to still do the same thing. The only thing that has changed for sensor configs 1 and 2 is the rate at which odometry is published. As discovered in the MARV model, each of the 5 diffdrive plugins published 50 Hz odometry to the same topic, so there were 250 messages per simulated second. I removed the 4 superfluous publishers and left only the one publishing odometry of the main tracks, so now the odometry topic has rate of 50 Hz.

### CODE REVIEW

**Comment by peci1** [comment]
absolem_all
2021-03-31T05:37:07 540955220

**Comment by peci1** [comment]
I force-pushed a SDF 1.6 version of Absolem SC3-5 based on #877 and #878. The geometry and behavior is exactly the same as in the original version. SC1-2 are "functionally" untouched.

**Comment by peci1** [comment]
I've resolved the conflicts brought in by #551 .

File: submitted_models/ctu_cras_norlab_absolem_sensor_config_1/urdf/boson_thermocam.xacro [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            <collision>
+                <origin xyz="${0.043/2 - 0.006} 0 0" />
+                <geometry>
+                    <box size="0.043 0.021 0.021"/>
+                </geometry>
+            </collision>
+            <xacro:box_inertial mass="${thermocam_mass}" width="0.021" height="0.021" depth="0.043" xyz="${0.043/2 - 0.006} 0 0" />
+        </link>
+        <xacro:if value="${simulate}">
+            <gazebo reference="$(arg prefix)${name}">
```
Comment (acschang) [inline submitted_models/ctu_cras_norlab_absolem_sensor_config_1/urdf/boson_thermocam.xacro:23]: Please adjust the thermal camera parameters to be consistent with the [updated values](https://github.com/osrf/subt/blob/ba0f2ec00e1db43197da0faf824cfda0ff8e6813/submitted_models/ctu_cras_norlab_marv_sensor_config_1/urdf/boson_thermocam.xacro) in #910.

**Review by acschang — CHANGES_REQUESTED** [review]
The initial assessment of these sensor configurations for the Absolem model is complete, however there are some issues to resolve before these additional sensor configurations can be merged and utilized in a competitive setting:

- [x] Please add all missing components to the RViz visualization.
- [ ] Please adjust the range resolution of the OS0-128 sensor to 0.01 to meet precedent.
- [x] Please adjust the thermal camera parameters to be consistent with the [updated values](https://github.com/osrf/subt/blob/ba0f2ec00e1db43197da0faf824cfda0ff8e6813/submitted_models/ctu_cras_norlab_marv_sensor_config_1/urdf/boson_thermocam.xacro) in #910.
- [ ] Please update the `specifications.md` file's `Sensors` section to include sensors in all sensor configurations.

Please address the issues noted above and add commits to this pull request as soon as possible. The submission is expected to be incorporated into the SubT Virtual Testbed pending a successful review.

**Comment by peci1** [comment]
Can you be more specific about the required changes in `specifications.md`? It seems to me that all sensor from all sensor configs are there... (lines 75-82 in https://github.com/ctu-vras/subt/blob/submitted_models/ctu_cras_norlab_absolem_improved/submitted_models/ctu_cras_norlab_absolem_sensor_config_1/specifications.md ).

**Comment by peci1** [comment]
I took care about the rest of the changes that were requested.

**Comment by acschang** [comment]
> Can you be more specific about the required changes in `specifications.md`? It seems to me that all sensors from all sensor configs are there... (lines 75-82 in https://github.com/ctu-vras/subt/blob/submitted_models/ctu_cras_norlab_absolem_improved/submitted_models/ctu_cras_norlab_absolem_sensor_config_1/specifications.md ).

I was originally thinking of adding all used in all configurations to [specifications.md Sensors section](https://github.com/ctu-vras/subt/blob/submitted_models/ctu_cras_norlab_absolem_improved/submitted_models/ctu_cras_norlab_absolem_sensor_config_1/specifications.md) as all robots are listed in the [Sensor Config Guide](https://github.com/ctu-vras/subt/blob/submitted_models/ctu_cras_norlab_absolem_improved/submitted_models/ctu_cras_norlab_absolem_sensor_config_1/specifications.md) section but upon second glance through the directory structure of additional sensor configurations for the Absolem model I believe the documentation is all present and follows the submission guidelines.

**Review by acschang — APPROVED** [review]

**Review by nkoenig — APPROVED** [review]

**Pull request merged**